### ISSUE TITLE
value_precede_chain, missing solution

### ISSUE THREAD

**Issue by matsc-at-sics-se** [chuffed/chuffed#147]
MiniZinc 2.7.5 (Chuffed 0.12.0).
The following model has 6 solutions, but Chuffed misses one, `[A,B,C] = [-3,2,1]`.
```
include "globals.mzn";
var -3.. -3 union 1..1: A;
var -2.. -2 union 2..2: B;
var -4.. -4 union -2.. -2 union 1..1: C;
constraint
  value_precede_chain([1,-2,-4],[A,B,A,C]);
solve :: int_search([C,A,B], input_order, indomain_min, complete) satisfy;
output ["[A,B,C] = \([A,B,C])\n"];
```

**Comment by Dekker1** [comment]
Hopefully this will be the same bug as #146, since `value_precede_chain` is rewritten into `seq_precede_chain`.

**Comment by Dekker1** [comment]
It actually appears the problems are separate. This issues should be resolved once #148 is merged
